ts/src/oxfun.ts

```diff
@@ -0,0 +1,2843 @@
+            const rawOrder = orders[i];
+            const symbol = this.safeString (rawOrder, 'symbol');
+            const market = this.market (symbol);
+            const type = this.safeString (rawOrder, 'type');
+            const side = this.safeString (rawOrder, 'side');
+            const amount = this.safeNumber (rawOrder, 'amount');
+            const price = this.safeNumber (rawOrder, 'price');
+            const orderParams = this.safeDict (rawOrder, 'params', {});
+            const orderRequest = this.createOrderRequest (market, type, side, amount, price, orderParams);
+            ordersRequests.push (orderRequest);
+        }
+        const request: Dict = {
+            'responseType': 'FULL',
+            'timestamp': this.milliseconds (),
+            'orders': ordersRequests,
+        };
+        const response = await this.privatePostV3OrdersPlace (this.extend (request, params));
+        const data = this.safeList (response, 'data', []);
+        return this.parseOrders (data);
+    }
+
+    createOrderRequest (market, type, side, amount, price = undefined, params = {}) {
+        /**
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} type 'market', 'limit', 'STOP_LIMIT' or 'STOP_MARKET'
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} amount how much of currency you want to trade in units of base currency
+         * @param {float} [price] the price at which the order is to be fullfilled, in units of the quote currency, ignored in market orders
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {int} [params.clientOrderId] a unique id for the order
+         * @param {float} [params.cost] the quote quantity that can be used as an alternative for the amount for market buy orders
+         * @param {float} [params.triggerPrice] The price at which a trigger order is triggered at
+         * @param {float} [params.limitPrice] Limit price for the STOP_LIMIT order
+         * @param {bool} [params.postOnly] if true, the order will only be posted if it will be a maker order
+         * @param {string} [params.timeInForce] GTC (default), IOC, FOK, PO, MAKER_ONLY or MAKER_ONLY_REPRICE (reprices order to the best maker only price if the specified price were to lead to a taker trade)
+         * @param {string} [params.selfTradePreventionMode] NONE, EXPIRE_MAKER, EXPIRE_TAKER or EXPIRE_BOTH for more info check here {@link https://docs.ox.fun/?json#self-trade-prevention-modes}
+         * @param {string} [params.displayQuantity] for an iceberg order, pass both quantity and displayQuantity fields in the order request
+         */
+        // const symbol = market['symbol'];
+        const request: Dict = {
+            'marketCode': market['id'],
+            'side': side.toUpperCase (),
+        };
+        const cost = this.safeNumber2 (params, 'cost', 'amount');
```

Comment:
we should never use safeNumber to construct the request because safeNUmber return type is dynamic, ideally we handle them as strings and at he end we convert it back if needed (most exchanges accept strings)

Reply:
Got it, sorry. We'll change it.